### DISCUSSION TITLE
store on sdcard

### DISCUSSION THREAD

**Discussion by tobiasBora** [nix-community/nix-on-droid#210] [Q&A]
Hello,

First thanks for this great software. However as I am always fighting for space I wanted to use my sdcard to store `/nix/store`. Since sdcards can't use ext4, I guess it would require to put it into a .img/qcow2 file, like what is done to run debian on Android.

Do you think it could be possible?

**Comment by t184256** [comment]
If your device offers a "move to storage card" button in Settings - Apps - Nix - Storage", use that.

Otherwise, if your device is rooted: yes, you can mount `/data/data/com.termux.nix/files/` from another location. You can create a separate ext4 partition on your SD card, or use a loopback mount with a file (what you call `.img`?). `QCOW2` is not relevant here, long story short, it's for virtual machines.

Otherwise, if none of the above is possible, some devices offer "merge storage" functionality, or whatever it's called when it combines internal storage with SD card, at the cost of requiring you to always use that specific SD card. This doesn't require rooting.

**Reply by tobiasBora** [reply]
Thanks for your answer!

Regarding the "move to storage" option, it is unfortunately not available on android 6.0 and onward, and I'm using Android 11 (some manufacturer tried to maintain it for a few more releases… but seems like most of them [stopped doing that](https://www.sony-mea.com/en/electronics/support/articles/SX658701)).

Unfortunately my phone is not rooted anymore, as I don't want to loose some functionalities like updates/pay by phone/… I was thinking that I may try to use `adb mount` to mount the file, but it [does not work either](https://android.stackexchange.com/questions/249009/mount-a-loopback-device-from-sd-card-without-root) and [I don't know how to apply that at each reboot](https://stackoverflow.com/questions/74233822/adb-loopback-mount-at-startup?noredirect=1#comment131062083_74233822).

Regarding merge storage, my device (a samsung) does not propose this from the interface and I was worried to turn all my SD card into internal storage… I saw that it was [still possible to do it via adb](https://forum.xda-developers.com/t/make-your-sd-card-semi-adopted-works-great.3399510/) even to use the sd card as both internal and external storage… but for some reasons when I reboot the external storage is smaller… but [the internal memory as not grown up](https://forum.xda-developers.com/t/samsung-xcover-cant-turn-sd-external-memory-into-internal-memory-even-with-adb.4511477/).:
```
# enable debug mode
$ adb shell
$ sm list-disks adoptable                                                                                                                                                                           
disk:179,32
$ sm partition disk:179,32 mixed 75 # 75% as external… make sure to backup data sd card will be formated
```
